arbnode/transaction_streamer.go

```diff
@@ -952,9 +952,7 @@ func (s *TransactionStreamer) WriteMessageFromSequencer(
 	if err := s.ExpectChosenSequencer(); err != nil {
 		return err
 	}
-	if !s.insertionMutex.TryLock() {
-		return execution.ErrSequencerInsertLockTaken
-	}
+	s.insertionMutex.Lock()
```

Comment:
A batch from inbox-reader could cause a reorg
This would be ReorgToAndEndBatch in TransactionStreamer, catching insertionMutex, then reorgMutex, then calling executionEngine's Reorg which catches createBlocksMutex

At the same time, the sequencer might be trying to sequence transactions .
SequenceTransactions calls sequencerWrapper. which catches createBlocksMutex and then calls WriteMessageFromSequencer.

So if WriteMessageFromSequencer catches insertionMutex - we arrive at a deadlock.

What we really need here is a way to attempt acquiring the lock with a timeout - enough to make tests pass, but with a backoff to prevent deadlocks

Reply:
Wouldn't the SequenceTransactions wait on createBlocksMutex to be released by the reorg though? createBlocksMutex is intentionally not released until after the new messages have been written